clearnode/channel_service.go

```diff
@@ -103,76 +103,110 @@ func (s *ChannelService) RequestCreate(wallet common.Address, params *CreateChan
 }
 
 func (s *ChannelService) RequestResize(params *ResizeChannelParams, rpcSigners map[string]struct{}, logger Logger) (ChannelOperationResponse, error) {
-	channel, err := GetChannelByID(s.db, params.ChannelID)
-	if err != nil {
-		logger.Error("failed to find channel", "error", err)
-		return ChannelOperationResponse{}, RPCErrorf("failed to find channel: %s", params.ChannelID)
-	}
-	if channel == nil {
-		return ChannelOperationResponse{}, RPCErrorf("channel %s not found", params.ChannelID)
-	}
+	var channel *Channel
+	var allocations []nitrolite.Allocation
+	err := s.db.Transaction(func(tx *gorm.DB) error {
+		var err error
+		channel, err = GetChannelByID(tx, params.ChannelID)
+		if err != nil {
+			logger.Error("failed to find channel", "error", err)
+			return RPCErrorf("failed to find channel: %s", params.ChannelID)
+		}
+		if channel == nil {
+			return RPCErrorf("channel %s not found", params.ChannelID)
+		}
 
-	if err = checkChallengedChannels(s.db, channel.Wallet); err != nil {
-		logger.Error("failed to check challenged channels", "error", err)
-		return ChannelOperationResponse{}, err
-	}
+		if err = checkChallengedChannels(tx, channel.Wallet); err != nil {
+			logger.Error("failed to check challenged channels", "error", err)
+			return err
+		}
 
-	if channel.Status != ChannelStatusOpen {
-		return ChannelOperationResponse{}, RPCErrorf("channel %s is not open: %s", params.ChannelID, channel.Status)
-	}
+		if channel.Status == ChannelStatusResizing {
+			return RPCErrorf("operation denied: resize already ongoing. Please complete the resize or close the channel %s", params.ChannelID)
+		}
 
-	_, ok := rpcSigners[channel.Wallet]
-	if !ok {
-		return ChannelOperationResponse{}, RPCErrorf("invalid signature")
-	}
+		if channel.Status != ChannelStatusOpen {
+			return RPCErrorf("channel %s is not open: %s", params.ChannelID, channel.Status)
```

Comment:
Let's be consistent with error messages.

```suggestion
			return RPCErrorf("operation denied: channel %s is not open: %s", params.ChannelID, channel.Status)
```